test/e2e/specs/infrastructure/infrastructure__experimental-gutenberg.ts

```diff
@@ -81,7 +81,21 @@ describe( DataHelper.createSuiteTitle( 'Gutenberg: Experimental Features' ), fun
 		const expectedBlockPatternCount = 50;
 		const frame = await editorPage.getEditorHandle();
 		const actualBlockPatternCount = await frame.evaluate(
-			`window.wp.data.select( 'core' ).getBlockPatterns().length`
+			() =>
+				new Promise( ( resolve ) => {
+					const wait = setInterval( () => {
+						// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+						// @ts-ignore
+						const patterns = window.wp.data.select( 'core' ).getBlockPatterns();
+						if ( patterns.length > 0 ) {
+							clearInterval( wait );
+							// This needs to be done in a loop until patterns
+							// request returns anything as initially it returns
+							// an empty array, making this case a false positive.
+							resolve( patterns.length );
+						}
```

Comment:
**suggestion (non-blocking)**: Do we want to put a shorter timeout limit on how long we'll wait for the patterns before failing?

I tested around with this, and if the patterns don't actually ever load, this takes the full duration of the _jest_ timeout (which is usually two minutes) to time out and fail. That feels a bit long to me. Maybe we can track how many intervals we've done and eventually reject?

Again, not necessary, but an idea to make this fail faster!

Reply:
Ah, I didn't realize the full timeout is 2 minutes! It is indeed way too long, then. I'd make it reject after a specific time, though. The successful request usually needs a bit under 1s to complete, so I'd say that 10s would be more than enough for it to timeout. What do you think?

Reply:
Updated in https://github.com/Automattic/wp-calypso/pull/68662/commits/d3b9a5b6697f08bc4f61be59307f2c7fb622f402!